### PR TITLE
Fix!: Postgres queue sweep

### PR DESCRIPTION
**Pull request by vchan**
Fix the Postgres queue's sweep to match the Redis queue

### CODE REVIEW

File: saq/queue/postgres.py [hunk 1]
```diff
@@ -306,11 +305,42 @@ async def sweep(self, lock: int = 60, abort: float = 5.0) -> list[str]:
             cursor.nextset()
             cursor.nextset()
             results = await cursor.fetchall()
-            for result in results:
-                job = self.deserialize(result[0])
+            for key, job_bytes, objid in results:
+                job = self.deserialize(job_bytes)
                 if not job:
```
Comment (tobymao) [inline saq/queue/postgres.py:310]: how can job be null?

File: saq/queue/postgres.py [hunk 2]
```diff
@@ -306,11 +305,42 @@ async def sweep(self, lock: int = 60, abort: float = 5.0) -> list[str]:
...
+                                DELETE FROM {jobs_table}
+                                WHERE key = %(key)s
+                                """
+                            )
+                        ).format(jobs_table=self.jobs_table),
+                        {"key": key},
+                    )
+                    continue
+                if job.status == Status.QUEUED:
+                    await asyncio.sleep(0.1)
```
Comment (tobymao) [inline saq/queue/postgres.py:326]: we can avoid this hack if there's a lock between dequeue and processing can we make that happen easily in postgres?

File: saq/queue/postgres.py (hunk 1 above)
Comment (vchan) [reply saq/queue/postgres.py:310]: It's theoretically possible the job column could be null, but there's no code path to that situation. Maybe if someone logs into the database and updates the job table manually. Want me to remove this? And/or add a not null constraint on the job column?

File: saq/queue/postgres.py (hunk 2 above)
Comment (vchan) [reply saq/queue/postgres.py:326]: I think the Postgres queue should already already handle it. I can probably remove the hack. When a job is dequeued, the job table's status column is updated to active in the same transaction by the queue, but the worker is the one that updates the status of the job object (separate transaction). The Postgres queue also grabs an advisory lock associated with the job though, and if the worker/queue's connection dies, the lock is released. The sweeper already takes care of active jobs without advisory locks.

File: saq/queue/postgres.py (hunk 1 above)
Comment (tobymao) [reply saq/queue/postgres.py:310]: yea sure, let's make job not null

File: saq/queue/postgres.py [hunk 3]
```diff
@@ -613,10 +620,11 @@ async def _dequeue(self) -> list[Job]:
...
             )
             results = await cursor.fetchall()
-        for result in results:
-            job = self.deserialize(result[0])
-            if job:
-                jobs.append(job)
+            for result in results:
+                job = self.deserialize(result[0])
+                if job:
+                    await self.update(job, status=Status.ACTIVE, connection=conn)
```
Comment (vchan) [inline saq/queue/postgres.py:626]: Update the job's status to match the table column status in the same transaction to keep them in sync.

**Review by tobymao — APPROVED** [review]

**Pull request merged**